In [2]:
import pandas as pd 

Read this dataset into a DataFrame named hospitals.

In [3]:
hospitals = pd.read_csv('../data/Hospitals.csv')

for data path: .. will move back one folder. use tab to autofill and select folders

Look at the first few rows and then look at the last few. What do you notice?

In [4]:
hospitals.head()

X             Y  OBJECTID        ID  \
0 -1.331889e+07  4.346975e+06         1   5793230   
1 -1.322651e+07  4.049626e+06         2  53391362   
2 -1.315620e+07  4.031978e+06         3  11190023   
3 -1.317190e+07  4.041752e+06         4  17090028   
4 -1.313208e+07  4.037270e+06         5  23691706   

                                                NAME  \
0                    CENTRAL VALLEY GENERAL HOSPITAL   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...   
2                  EAST LOS ANGELES DOCTORS HOSPITAL   
3          SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                      KINDRED HOSPITAL BALDWIN PARK   

                     ADDRESS             CITY STATE    ZIP           ZIP4  \
0    1025 NORTH DOUTY STREET          HANFORD    CA  93230  NOT AVAILABLE   
1             150 VIA MERIDA  WESTLAKE VILAGE    CA  91362  NOT AVAILABLE   
2    4060 WHITTIER BOULEVARD      LOS ANGELES    CA  90023  NOT AVAILABLE   
3     6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  90028  NOT AVAILABLE   
4  14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  91706  NOT AVAILABLE   

   ...             VAL_DATE                           WEBSITE       STATE_ID  \
0  ...  2014/02/10 00:00:00      http://www.hanfordhealth.com  NOT AVAILABLE   
1  ...  2014/02/10 00:00:00  http://www.losrobleshospital.com  NOT AVAILABLE   
2  ...  2014/02/10 00:00:00             http://www.elalax.com  NOT AVAILABLE   
3  ...  2014/02/10 00:00:00         http://sch-hollywood.com/  NOT AVAILABLE   
4  ...  2014/02/10 00:00:00      http://www.khbaldwinpark.com  NOT AVAILABLE   

                                    ALT_NAME ST_FIPS        OWNER TTL_STAFF  \
0                              NOT AVAILABLE       6  PROPRIETARY      -999   
1                              NOT AVAILABLE       6  PROPRIETARY      -999   
2                              NOT AVAILABLE       6  PROPRIETARY      -999   
3  HOLLYWOOD COMMUNITY HOSPITAL OF HOLLYWOOD       6  PROPRIETARY      -999   
4                              NOT AVAILABLE       6  PROPRIETARY      -999   

   BEDS         TRAUMA  HELIPAD  
0    49  NOT AVAILABLE        N  
1    62  NOT AVAILABLE        N  
2   127  NOT AVAILABLE        N  
3   100  NOT AVAILABLE        N  
4    95  NOT AVAILABLE        N  

[5 rows x 34 columns]

In [5]:
hospitals.tail(2)

X             Y  OBJECTID         ID  \
7594 -8.522423e+06  4.718512e+06      7610  191021401   
7595 -1.244921e+07  4.976468e+06      7611  196784108   

                                    NAME                     ADDRESS  \
7594  J KENT MCNEW FAMILY MEDICAL CENTER  175 HARRY S TRUMAN PARKWAY   
7595       UNIVERSITY ORTHOPAEDIC CENTER              590 WAKARA WAY   

                CITY STATE    ZIP           ZIP4  ...             VAL_DATE  \
7594       ANNAPOLIS    MD  21401  NOT AVAILABLE  ...  2020/05/05 00:00:00   
7595  SALT LAKE CITY    UT  84108  NOT AVAILABLE  ...  2020/05/05 00:00:00   

                                                WEBSITE       STATE_ID  \
7594  https://aahs.org/locations/j--kent-mcnew-famil...  NOT AVAILABLE   
7595          https://healthcare.utah.edu/orthopaedics/  NOT AVAILABLE   

           ALT_NAME ST_FIPS               OWNER TTL_STAFF  BEDS  \
7594  NOT AVAILABLE      24          NON-PROFIT      -999  -999   
7595  NOT AVAILABLE      49  GOVERNMENT - STATE      -999  -999   

             TRAUMA  HELIPAD  
7594  NOT AVAILABLE        N  
7595  NOT AVAILABLE        N  

[2 rows x 34 columns]

Observations: lots of data, unsure what all of the columns mean

How many rows and columns does this DataFrame have?

In [6]:
hospitals.shape

(7596, 34)

In [7]:
hospitals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7596 entries, 0 to 7595
Data columns (total 34 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   X           7596 non-null   float64
 1   Y           7596 non-null   float64
 2   OBJECTID    7596 non-null   int64  
 3   ID          7596 non-null   int64  
 4   NAME        7596 non-null   object 
 5   ADDRESS     7596 non-null   object 
 6   CITY        7596 non-null   object 
 7   STATE       7596 non-null   object 
 8   ZIP         7596 non-null   int64  
 9   ZIP4        7596 non-null   object 
 10  TELEPHONE   7596 non-null   object 
 11  TYPE        7596 non-null   object 
 12  STATUS      7596 non-null   object 
 13  POPULATION  7596 non-null   int64  
 14  COUNTY      7596 non-null   object 
 15  COUNTYFIPS  7596 non-null   object 
 16  COUNTRY     7596 non-null   object 
 17  LATITUDE    7596 non-null   float64
 18  LONGITUDE   7596 non-null   float64
 19  NAICS_CODE  7596 non-null  

We don't need all of the columns. Slice the dataframe down so that it only contains the following columns: 'NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE', 'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS'

In [8]:
hospitals[['NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE', 'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS']]

NAME  \
0                       CENTRAL VALLEY GENERAL HOSPITAL   
1     LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...   
2                     EAST LOS ANGELES DOCTORS HOSPITAL   
3             SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                         KINDRED HOSPITAL BALDWIN PARK   
...                                                 ...   
7591      ADVENTIST HEALTHCARE WHITE OAK MEDICAL CENTER   
7592          COPPER QUEEN DOUGLAS EMERGENCY DEPARTMENT   
7593                          WHITING FORENSIC HOSPITAL   
7594                 J KENT MCNEW FAMILY MEDICAL CENTER   
7595                      UNIVERSITY ORTHOPAEDIC CENTER   

                         ADDRESS             CITY STATE                TYPE  \
0        1025 NORTH DOUTY STREET          HANFORD    CA  GENERAL ACUTE CARE   
1                 150 VIA MERIDA  WESTLAKE VILAGE    CA  GENERAL ACUTE CARE   
2        4060 WHITTIER BOULEVARD      LOS ANGELES    CA  GENERAL ACUTE CARE   
3         6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  GENERAL ACUTE CARE   
4      14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  GENERAL ACUTE CARE   
...                          ...              ...   ...                 ...   
7591           11890 HEALING WAY    SILVER SPRING    MD  GENERAL ACUTE CARE   
7592           100 E. 5TH STREET          DOUGLAS    AZ  GENERAL ACUTE CARE   
7593                70 OBRIEN DR       MIDDLETOWN    CT         PSYCHIATRIC   
7594  175 HARRY S TRUMAN PARKWAY        ANNAPOLIS    MD         PSYCHIATRIC   
7595              590 WAKARA WAY   SALT LAKE CITY    UT  GENERAL ACUTE CARE   

      POPULATION        COUNTY COUNTYFIPS  \
0             49         KINGS      06031   
1             62       VENTURA      06111   
2            127   LOS ANGELES      06037   
3            100   LOS ANGELES      06037   
4             95   LOS ANGELES      06037   
...          ...           ...        ...   
7591         178    MONTGOMERY      24031   
7592        -999       COCHISE      04003   
7593         229     MIDDLESEX      09007   
7594        -999  ANNE ARUNDEL      24003   
7595        -999     SALT LAKE      49035   

                                     NAICS_DESC  BEDS  
0        GENERAL MEDICAL AND SURGICAL HOSPITALS    49  
1        GENERAL MEDICAL AND SURGICAL HOSPITALS    62  
2        GENERAL MEDICAL AND SURGICAL HOSPITALS   127  
3        GENERAL MEDICAL AND SURGICAL HOSPITALS   100  
4        GENERAL MEDICAL AND SURGICAL HOSPITALS    95  
...                                         ...   ...  
7591     GENERAL MEDICAL AND SURGICAL HOSPITALS   178  
7592     GENERAL MEDICAL AND SURGICAL HOSPITALS  -999  
7593  PSYCHIATRIC AND SUBSTANCE ABUSE HOSPITALS   229  
7594  PSYCHIATRIC AND SUBSTANCE ABUSE HOSPITALS  -999  
7595     GENERAL MEDICAL AND SURGICAL HOSPITALS  -999  

[7596 rows x 10 columns]

In [9]:
hospitals = hospitals[['NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE', 'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS']]

What are the different hospital types (contained in the TYPE column)? Which is the most common?

In [10]:
hospitals['TYPE'].unique()

array(['GENERAL ACUTE CARE', 'PSYCHIATRIC', 'CHILDREN', 'LONG TERM CARE',
       'CRITICAL ACCESS', 'REHABILITATION', 'MILITARY', 'WOMEN',
       'SPECIAL', 'CHRONIC DISEASE'], dtype=object)

In [11]:
hospitals['TYPE'].nunique()

10

In [12]:
hospitals['TYPE'].value_counts()

GENERAL ACUTE CARE    4481
CRITICAL ACCESS       1027
PSYCHIATRIC            759
LONG TERM CARE         431
REHABILITATION         378
MILITARY               216
SPECIAL                145
CHILDREN               130
WOMEN                   20
CHRONIC DISEASE          9
Name: TYPE, dtype: int64

Highest count: general acute care

Find all hospitals in Nashville, Tennessee. Take a look at the BEDS column. What do you notice? Why do you think this is?

In [21]:
hospitals.loc[(hospitals['CITY'] == 'NASHVILLE') & (hospitals['STATE'] =='TN')]

NAME  \
545                  TRISTAR SKYLINE MEDICAL CENTER   
3449          SELECT SPECIALTY HOSPITAL - NASHVILLE   
3485                           CURAHEALTH NASHVILLE   
5429             THE WOMEN'S HOSPITAL AT CENTENNIAL   
5433        METROPOLITAN NASHVILLE GENERAL HOSPITAL   
6289                  SAINT THOMAS MIDTOWN HOSPITAL   
6293  VANDERBILT STALLWORTH REHABILITATION HOSPITAL   
6661           VA MEDICAL CENTER - NASHVILLE CAMPUS   
6774    SAINT THOMAS HOSPITAL FOR SPECIALTY SURGERY   
6777                     SAINT THOMAS WEST HOSPITAL   
6786          TRISTAR SOUTHERN HILLS MEDICAL CENTER   
6808              TRISTAR CENTENNIAL MEDICAL CENTER   
6813                 VANDERBILT UNIVERSITY HOSPITAL   

                             ADDRESS       CITY STATE                TYPE  \
545              3441 DICKERSON PIKE  NASHVILLE    TN  GENERAL ACUTE CARE   
3449    2000 HAYES STREET, SUITE 500  NASHVILLE    TN      LONG TERM CARE   
3485  1412 COUNTY HOSPITAL ROAD, B-1  NASHVILLE    TN      LONG TERM CARE   
5429              2221 MURPHY AVENUE  NASHVILLE    TN               WOMEN   
5433              1818 ALBION STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6289              2000 CHURCH STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6293             2201 CHILDREN'S WAY  NASHVILLE    TN      REHABILITATION   
6661                 1310 24TH AVE S  NASHVILLE    TN            MILITARY   
6774              2011 MURPHY AVENUE  NASHVILLE    TN  GENERAL ACUTE CARE   
6777               4220 HARDING ROAD  NASHVILLE    TN  GENERAL ACUTE CARE   
6786                391 WALLACE ROAD  NASHVILLE    TN  GENERAL ACUTE CARE   
6808           2300 PATTERSON STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6813       1211 MEDICAL CENTER DRIVE  NASHVILLE    TN  GENERAL ACUTE CARE   

      POPULATION    COUNTY COUNTYFIPS  \
545          213  DAVIDSON      47037   
3449          70  DAVIDSON      47037   
3485          60  DAVIDSON      47037   
5429        -999  DAVIDSON      47037   
5433         150  DAVIDSON      47037   
6289         683  DAVIDSON      47037   
6293          80  DAVIDSON      47037   
6661         493  DAVIDSON      47037   
6774          23  DAVIDSON      47037   
6777         541  DAVIDSON      47037   
6786         126  DAVIDSON      47037   
6808         733  DAVIDSON      47037   
6813        1091  DAVIDSON      47037   

                                             NAICS_DESC  BEDS  
545              GENERAL MEDICAL AND SURGICAL HOSPITALS   213  
3449  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...    70  
3485  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...    60  
5429  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...  -999  
5433             GENERAL MEDICAL AND SURGICAL HOSPITALS   150  
6289             GENERAL MEDICAL AND SURGICAL HOSPITALS   683  
6293  REHABILITATION HOSPITALS (EXCEPT ALCOHOLISM, D...    80  
6661             GENERAL MEDICAL AND SURGICAL HOSPITALS   493  
6774             GENERAL MEDICAL AND SURGICAL HOSPITALS    23  
6777             GENERAL MEDICAL AND SURGICAL HOSPITALS   541  
6786             GENERAL MEDICAL AND SURGICAL HOSPITALS   126  
6808             GENERAL MEDICAL AND SURGICAL HOSPITALS   733  
6813             GENERAL MEDICAL AND SURGICAL HOSPITALS  1091

How to sort by both city AND state? You can satisfy or '|' and and '&' 
Beds with -999 == NaN value? 

In [31]:
hospitals.loc[hospitals['STATE'] == 'TN'] ['COUNTY'].value_counts()

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON     6
              ..
TROUSDALE      1
HOUSTON        1
FAYETTE        1
HENRY          1
FENTRESS       1
Name: COUNTY, Length: 80, dtype: int64

In [32]:
TN_hospitals = hospitals.loc[hospitals['STATE'] == 'TN']

Which county in TN has the most hospitals?: Shelby

In [23]:
TN_hospitals['COUNTY'].value_counts()

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON     6
              ..
TROUSDALE      1
HOUSTON        1
FAYETTE        1
HENRY          1
FENTRESS       1
Name: COUNTY, Length: 80, dtype: int64

There are 95 counties in Tennessee. How many of them have no hospitals at all?
15=95-80 from list above

In [37]:
TN_hospitals.shape

(171, 10)